### PR TITLE
Handle missing channels in `scan.get_image("rgb")` gracefully

### PR DESCRIPTION
**Pull request by tobiasjj**
**Why this PR?**
Missing channels in `scan.rgb_image` should be handled gracefully so that users still could see remaining channels.

### CODE REVIEW

**Review by JoepVanlier — CHANGES_REQUESTED** [review]
Cool stuff! It'll be really nice to have this in. This would mean that we can also trim down files, because we don't need to have all channels in anymore if we're only using red for instance.

Great work on the tests too! I have a few suggestions/requests, which might simplify the code a little bit.

Oh, one small note, you don't need to put the ticket number in the branch or description, since these numbers can't be seen from the outside and have little meaning to outside viewers.

File: lumicks/pylake/tests/test_imaging_confocal/test_scan.py [hunk 1]
```diff
@@ -346,15 +378,19 @@ def test_scan_plot_single_channel_absolute_color_adjustment(test_scans):
...
         plt.close(fig)
 
         # Test whether setting a single color works correctly (should use the same for R G and B).
         fig = plt.figure()
         scan.plot(channel=channel, adjustment=ColorAdjustment(lb, ub, mode="absolute"))
         image = plt.gca().get_images()[0]
-        np.testing.assert_allclose(image.get_array(), scan.get_image(channel)) #getattr(scan, f"{channel}_image"))
+        np.testing.assert_allclose(
+            image.get_array(), scan.get_image(channel)
+        )  # getattr(scan, f"{channel}_image"))
```
[the remark this answers is not in the document]
Comment (tobiasjj) [reply lumicks/pylake/tests/test_imaging_confocal/test_scan.py:393]: Indeed, they are.

File: lumicks/pylake/tests/test_imaging_confocal/conftest.py [hunk 2]
```diff
@@ -185,6 +185,45 @@ def test_scans(reference_counts):
     )
     scans["multiframe_poisson"] = Scan("multiframe_poisson", mock_file, start, stop, metadata)
 
+    mock_file, metadata, stop = MockConfocalFile.from_streams(
+        start,
```
[the remark this answers is not in the document]
Comment (tobiasjj) [reply lumicks/pylake/tests/test_imaging_confocal/conftest.py:189]: Thanks!

File: lumicks/pylake/detail/confocal.py [hunk 3]
```diff
@@ -488,6 +496,25 @@ def get_image(self, channel="rgb"):
...
         if channel == "rgb":
-            return np.stack([self.get_image(color).T for color in ("red", "green", "blue")]).T
+            # Get rgb images, check if images are empty
+            images = []
+            empty_image_idx = []
+            shape = None
+            for i, color in enumerate(("red", "green", "blue")):
+                image = self.get_image(color).T
+                images.append(image)
+                if shape is None or shape[0] == 0:
```
[the remark this answers is not in the document]
Comment (tobiasjj) [reply lumicks/pylake/detail/confocal.py:506]: Indeed.

File: lumicks/pylake/detail/confocal.py [hunk 4]
```diff
@@ -47,8 +48,10 @@ def timestamp_mean(a, axis=None):
 
 
 def _default_image_factory(self: "ConfocalImage", color):
-    channel_data = getattr(self, f"{color}_photon_count").data
-    raw_image = reconstruct_image_sum(channel_data.astype(float), self.infowave.data, self._shape)
+    channel = getattr(self, f"{color}_photon_count")
+    if channel == empty_slice:
+        return channel.data
```
[the remark this answers is not in the document]
Comment (tobiasjj) [reply lumicks/pylake/detail/confocal.py:53]: Will change it.

**Review by JoepVanlier — APPROVED** [review]
Looks good to me! :shipit:  :)

**Pull request merged**